src/Microsoft.Extensions.ML/ModelLoaders/FileModelLoader.cs

```diff
@@ -105,16 +105,39 @@ public override ITransformer GetModel()
             return _model;
         }
 
+        private FileStream WaitForFile(string fullPath, FileMode mode, FileAccess access, FileShare share)
+        {
+            for (int numTries = 0; numTries < 100; numTries++)
+            {
+                FileStream fs = null;
+                try
+                {
+                    fs = new FileStream(fullPath, mode, access, share);
+                    return fs;
+                }
+                catch (IOException)
+                {
+                    if (fs != null)
+                    {
+                        fs.Dispose();
+                    }
+                    Thread.Sleep(50);
+                }
+            }
+
+            return null;
+        }
+
         //internal virtual for testing purposes.
         internal virtual void LoadModel()
         {
-            //Sleep to avoid some file system locking issues
-            //TODO: The same thing occurs in configuration reload
-            //we should make sure the sleeps are the same.
-            Thread.Sleep(50);
-            using (var fileStream = File.OpenRead(_filePath))
+            var fs = WaitForFile(_filePath, FileMode.Open, FileAccess.Read, FileShare.Read);
+            if (fs == null)
```

Comment:
Since you have a repro for this issue, can you please check it in as a test for this bug?

Reply:
There is no easy way to test this, this repro requires to write a web api and call webapi in certain order and then check the output. We need to introduce some web test framework to do this.

Reply:
You don't need a web test to use `Microsoft.Extensions.ML`. You can use it from any test project. We already have some tests here:

https://github.com/dotnet/machinelearning/tree/master/test/Microsoft.Extensions.ML.Tests

Reply:
@eerhardt Thanks for the pointer. @frank-dong-ms Can you please add a test for this issue?

Reply:
Sure, thanks Eric, will do that in separate PR.